src/main/resources/db/changelog/20231003113535-entity_layering_schema.xml

```diff
@@ -354,7 +354,357 @@
         <dropForeignKeyConstraint baseTableName="cp_pool" constraintName="cp_pool_fk1"/>
     </changeSet>
 
-    <!-- TODO: Add migration logic here -->
+   <changeSet id="20231003113535-11" author="crog" dbms="postgresql, mysql, mariadb">
+        <sql>
+            -- Migrate global products and content
+            DROP TABLE IF EXISTS tmp_active_global_products;
+            CREATE TEMPORARY TABLE tmp_active_global_products (
+                uuid VARCHAR(32) NOT NULL,
+                pid VARCHAR(32) NOT NULL,
+                new_uuid VARCHAR(32) NOT NULL,
+                best BOOLEAN NOT NULL DEFAULT false,
+                PRIMARY KEY (uuid));
+
+            DROP TABLE IF EXISTS tmp_active_global_contents;
+            CREATE TEMPORARY TABLE tmp_active_global_contents (
+                uuid VARCHAR(32) NOT NULL,
+                cid VARCHAR(32) NOT NULL,
+                new_uuid VARCHAR(32) NOT NULL,
+                best BOOLEAN NOT NULL DEFAULT false,
+                PRIMARY KEY (uuid));
+
+            -- Collect "active" global products and their children
+            INSERT INTO tmp_active_global_products (uuid, pid, new_uuid, best)
+                WITH RECURSIVE parent_child_map(parent_uuid, child_uuid) AS (
+                    SELECT uuid AS parent_uuid, derived_product_uuid AS child_uuid
+                        FROM cp2_products
+                        WHERE derived_product_uuid IS NOT NULL
+                    UNION
+                    SELECT product_uuid AS parent_uuid, provided_product_uuid AS child_uuid
+                        FROM cp2_product_provided_products),
+                product_graph(uuid, depth) AS (
+                    SELECT active.uuid, 0
+                        FROM cp2_products prod
+                        JOIN (SELECT product_uuid AS uuid FROM cp2_owner_products
+                            UNION
+                            SELECT product_uuid AS uuid FROM cp_pool) active ON active.uuid = prod.uuid
+                        WHERE prod.locked != 0
+                    UNION
+                    SELECT pcmap.child_uuid, pg.depth + 1
+                        FROM parent_child_map pcmap
+                        JOIN product_graph pg ON pg.uuid = pcmap.parent_uuid
+                        -- shouldn't be needed, but here just in case someone decided to be cheeky and define a
+                        -- cyclical product, or some other crazy thing that's entirely unsupported by CP's logic
+                        -- and queries
+                        WHERE pg.depth &lt; 9)
```

Comment:
is level 9 really needed here? isn't our current hierarchy a max depth of 2 (excluding contents, marketing product is 0, derived product is 1, provided product 2)?